front/components/app/CopyRun.tsx

```diff
@@ -0,0 +1,131 @@
+import "@uiw/react-textarea-code-editor/dist.css";
+
+import { Button, CubeIcon, Tooltip } from "@dust-tt/sparkle";
+import type { WorkspaceType } from "@dust-tt/types";
+import type { AppType, SpecificationType } from "@dust-tt/types";
+import type { RunType } from "@dust-tt/types";
+import { Dialog, Transition } from "@headlessui/react";
+import { Fragment, useMemo, useState } from "react";
+
+import { DisplayCurlRequest } from "@app/components/app/Deploy";
+import { useRunBlock } from "@app/lib/swr";
+
+interface CopyRunProps {
+  app: AppType;
+  disabled: boolean;
+  owner: WorkspaceType;
+  run: RunType;
+  spec: SpecificationType;
+  url: string;
+}
+
+export default function CopyRun({
+  app,
+  disabled,
+  owner,
+  run,
+  spec,
+  url,
+}: CopyRunProps) {
+  const [open, setOpen] = useState(false);
+
+  const [firstBlock] = spec;
+
+  const { run: runDetails } = useRunBlock(
+    owner,
+    app,
+    run.run_id,
+    firstBlock.type,
+    firstBlock.name,
+    () => {
```

Comment:
nit:: hard to understand what this thing is because the params is not an object

Reply:
Yep agree. It was already there, it's used to change the refresh interval based on the run status.